Registry/prefab.editor.setreg

```diff
@@ -17,5 +17,12 @@
                 }
             }
         }
+    },
+    "O3DE": {
+        "Preferences": {
+            "Prefabs": {
+                "EnableOutlinerOverrideManagement": true
+            }
```

Comment:
By adding the default value here, the key/value would be present in `<project_root>/user/Registry/editorpreferences.setreg`, so users can change `true` to `false` if they want to disable the feature.

Reply:
Instead of plaching this setting the engine `Registry/prefab.editor.setreg` it could be placed in the `*.setreg` that is part of the [DefaultProject](https://github.com/o3de/o3de/tree/development/Templates/DefaultProject/Template/Registry) Template and [MinimalProject](https://github.com/o3de/o3de/tree/development/Templates/MinimalProject/Template/Registry)

Reply:
GP. That is also a place to consider. Another alternative is default the flag to `true` in code like we did for Action Manager and Prefab System.

https://github.com/o3de/o3de/blob/development/Code/Framework/AzFramework/AzFramework/Application/Application.cpp#L759

Will see what others think about these options.

Reply:
Regarding only storing it in DefaultProject and MinimalProject, 

1. Would it be possible to create a new project without using these 2 templates? If so, the value will be false by default
2. If we create a new template in the future like 'DefaultStarterGameProject' , then the creator of the template would need to remember and know that they need to enable the flag. Relying on project creators to know and remember this would be asking a lot imo.
3. Also, existing projects will still see the old 'non-override' mode. Ideally, we would want the 'override' mode to be the default for all projects and want users to switch to non-override mode if they run into issues (this can't be forever though) 

Considering the above reasons, I think it's better to either have a global setreg that flips it to true like shown here or change it in the code.

Reply:
For number 1 Creating a project without using either of those templates is an unsupported workflow. Someone could copy and paste an existing project and rename all the content within it to make a "new" project, but is off the beaten path

For number 2, we create project templates from existing projects. So if the user created a `DefaultStarterGameProject` template they would do so from a project created from the Default Project template.

For number 3, the default value in code should be `true`, so the need to have a engine setreg file or even a project setreg is unneeded.

Reply:
> For number 1 Creating a project without using either of those templates is an unsupported workflow. Someone could copy and paste an existing project and rename all the content within it to make a "new" project, but is off the beaten path.

I think we don't need to change the flag in the project template. If we do that, if I understand right we need to manually do the same every time a new project template is published (e.g. MultiplayerProject, FPSProject, etc), unless all other templates can inherit data from those two basic templates.

> For number 3, the default value in code should be true, so the need to have a engine setreg file or even a project setreg is unneeded.

Can we do both (code + global setreg file)?

Reply:
Due to time constraint, I need to get the PR in before stabilization cut on 3/15. I would go with the current version (global setreg file). Let me know if you still have any concern @srikappa-amzn, or I'll try to merge it before the time on 3/15.

Reply:
I'm fine with this going as is. The code change won't be needed if we are doing this.